File: lab02/Universidad.java
import java.util.*;
public class Universidad {
    private String nombre;
    private List<Curso> cursos;
    
    public Universidad(){
        this.cursos= new ArrayList<>();
    }
    public Universidad(String nombre){
        this.cursos= new ArrayList<>();
        this.nombre=nombre;
    }
    public String getNombre(){
        return nombre;
    }
    public void setNombre(String nombre){
        this.nombre=nombre;
    }
    public List<Curso> getCursos(){
        return cursos;
    }
    public void setCursos(List<Curso> cursos){
        this.cursos=cursos;
    }
    public void agregarCurso(Curso curso){
        this.cursos.add(curso);
    }
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Universidad: ").append(nombre).append("\nCursos ofrecidos:\n");
        for (Curso c : cursos) {
            sb.append(" - ").append(c.toString()).append("\n");
        }
        return sb.toString();
    }
}

File: lab02/Universidad.py
from typing import List
from Curso import Curso

class Universidad:

    def __init__(self, nombre: str = ""):
        self._nombre = nombre
        self._cursos: List[Curso] = []  # Agregación

    @property
    def nombre(self) -> str:
        return self._nombre

    @nombre.setter
    def nombre(self, valor: str):
        self._nombre = valor

    @property
    def cursos(self) -> List[Curso]:
        return self._cursos

    @cursos.setter
    def cursos(self, valor: List[Curso]):
        self._cursos = valor

    def agregar_curso(self, curso: Curso):
        self._cursos.append(curso)

    def __str__(self) -> str:
        salida = f"Universidad: {self._nombre}\nCursos ofrecidos:\n"
        for c in self._cursos:
            salida += f" - {c}\n"
        return salida